modular_skyrat/modules/cargo/code/strappedcrate.dm

```diff
@@ -0,0 +1,102 @@
+	if(ismovable(loc))
+		var/atom/movable/AM = loc //can't unwrap the wrapped container if it's inside something.
+		AM.relay_container_resist_act(user, strapped_object)
+		return
+	to_chat(user, "<span class='notice'>You lean on the back of [strapped_object] and start pushing to snap the straps off.</span>")
+	if(do_after(user, 150, target = strapped_object))  //takes slightly longer to break plastic as opposed to paper
+		if(!user || user.stat != CONSCIOUS || user.loc != strapped_object || strapped_object.loc != src )
+			return
+		to_chat(user, "<span class='notice'>You successfully broke [strapped_object]'s straps!</span>")
+		strapped_object.forceMove(loc)
+		playsound(src.loc, 'sound/items/poster_ripped.ogg', 50, TRUE)
+		new /obj/item/stack/sheet/plastic (loc, 4)
+		unwrap_contents()
+		qdel(src)
+	else
+		if(user.loc == src) //so we don't get the message if we resisted multiple times and succeeded.
+			to_chat(user, "<span class='warning'>You fail to remove [strapped_object]'s straps!</span>")
+
+//Lets plastic wrap crates and only crates, then overlays the plastic straps/locks it shut
+//Basically applies package wrapping code but for plastic onto crates
+/obj/item/stack/sheet/plastic/afterattack(obj/target, mob/user, proximity)
+	. = ..()
+	if(!proximity)
+		return
+	if(!istype(target))
+		return
+	if(target.anchored)
+		return
+
+	if(istype (target, /obj/structure/closet))	//this extra IF check is basically so that hitting EVERY item doesnt say 'you cant wrap this', and only hitting invalid storage items does
+		if(istype (target, /obj/structure/closet/crate))
+			var/obj/structure/closet/crate/tcrate = target
+			if(tcrate.opened)
+				return
+			if(!tcrate.delivery_icon) //no delivery icon means unwrappable crate
+				to_chat(user, "<span class='warning'>You can't strap this shut!</span>")
+				return
+			if(istype(tcrate, /obj/structure/closet/crate/coffin) || istype(tcrate, /obj/structure/closet/crate/trashcart) || istype(tcrate, /obj/structure/closet/crate/bin) || istype(tcrate, /obj/structure/closet/crate/critter) || istype(tcrate, /obj/structure/closet/crate/necropolis) || istype(tcrate, /obj/structure/closet/crate/miningcar))	//for buggy-wuggy and ugly-wugly crates
+				to_chat(user, "<span class='warning'>You can't strap this shut!</span>")
+				return
+			if(use(4))
+				if(!do_after(user, 20, target = user))	//For some reason this still deletes stacks of exactly four, even if you fail?
+					return
+				var/obj/structure/big_delivery/strapped/new_strap = new /obj/structure/big_delivery/strapped(get_turf(tcrate.loc))
+				new_strap.name = "strapped [tcrate.name]"	//you can still tell what crate is under the straps
+				new_strap.icon_state = tcrate.icon_state	//inherits the sprite of the crate it was used on
+				new_strap.overlays = tcrate.overlays //also inherits the overlays, for the sake of locks and papers. THIS OVERWRITES THE INITIAL CRATE-STRAP OVERLAY
```

Comment:
Yeah please make this a function of the crate itself, using crate update overlays and such...